python/tests/test_cdf.py

```diff
@@ -677,3 +677,40 @@ def test_write_overwrite_partitioned_cdf(tmp_path, sample_data: pa.Table):
     ).sort_by(sort_values).select(expected_data.column_names) == pa.concat_tables(
         [first_batch, expected_data]
     ).sort_by(sort_values)
+
+
+def test_read_cdf_version_out_of_range():
+    dt = DeltaTable("../crates/test/tests/data/cdf-table/")
+
+    try:
+        b = dt.load_cdf(4).read_all().to_pydict()
+        assert False, "Should not get here"
+    except _internal.DeltaError as e:
```

Comment:
You can do deltalake.exceptions actually

Reply:
I have changed it to use the exceptions module